### PR TITLE
HAWQ-779. support more pxf filter pushdwon

### PR DESCRIPTION
**Pull request by jiadexin**
https://issues.apache.org/jira/browse/HAWQ-779
Description
When I use the pxf hawq, I need to read a traditional relational database systems and solr by way of the external table. The project :https://github.com/Pivotal-Field-Engineering/pxf-field/tree/master/jdbc-pxf-ext, only "WriteAccessor ",so I developed 2 plug-ins, the projects: https://github.com/inspur-insight/pxf-plugin , But these two plug-ins need to modified HAWQ:
1. When get a list of fragment from pxf services, push down the 'filterString'. modify the backend / optimizer / plan / createplan.c of create_pxf_plan methods:
segdb_work_map = map_hddata_2gp_segments (uri_str,
total_segs, segs_participating,
relation, ctx-> root-> parse-> jointree-> quals);
2. modify pxffilters.h and pxffilters.c, support TEXT types LIKE operation, Date type data operator, Float type operator.
3. Modify org.apache.hawq.pxf.api.FilterParser.java, support the LIKE operator.

### CODE REVIEW

**Comment by shivzone** [comment]
Please update the corresponding test in FilterParserTest to include the new operators

File: src/backend/access/external/pxffilters.c [hunk 1]
```diff
@@ -125,7 +126,48 @@ dbop_pxfop_map pxf_supported_opr[] =
...
+
+	/* float8 */
+	{Float8EqualOperator  /* float8eq */, PXFOP_EQ},
+	{672  /* float8lt */, PXFOP_LT},
+	{674 /* float8gt */, PXFOP_GT},
+	{673 /* float8le */, PXFOP_LE},
+	{675 /* float8ge */, PXFOP_GE},
+	{671 /* float8ne */, PXFOP_NE},
+
+	/* float48 */
```
Comment (shivzone) [inline src/backend/access/external/pxffilters.c]: This pull request doesn't quite seem related to supporting operators on Float. What is the purpose of this ?

Comment (jiadexin) [inline src/backend/access/external/pxffilters.c]: When using HAWQ PXF read relational database, FLOAT is a very common type, and often used in conditional expressions. So I added FLOAT operators .

**Comment by jiadexin** [comment]
My local branch some confusion, handle submodules marked for deletion, I do not know how to recover. I first created PR in GITHUB, not familiar with.

**Comment by shivzone** [comment]
+1

**Comment by shivzone** [comment]
Merged this feature to master. Closing this pulll request

**Pull request closed without merging**